tests/testsuite/update.rs

```diff
@@ -2016,51 +1927,1940 @@ fn update_breaking_specific_packages() {
+    Package::new("shared", "0.2.1").publish();
+
+    p.cargo("update -Zupdate-precise-breaking -p shared@0.1.0 --precise 0.2.0")
+        .masquerade_as_nightly_cargo(&[])
+        .with_status(101)
+        // The lockfile update is failing for the same reason a non-breaking
+        // update would. There is still a shared@0.1.0 here that didn't get
+        // upgraded (the transitive dependency within bar), and we are now
+        // asking the lockfile update to update it to 0.2.0, which it cannot do.
         .with_stderr_data(str![[r#"
+[UPGRADING] shared ^0.1 -> ^0.2
 [UPDATING] `dummy-registry` index
-[UPGRADING] shared ^1.0 -> ^2.0
-[UPGRADING] ws ^1.0 -> ^2.0
-[UPGRADING] just-foo ^1.0 -> ^2.0
-[LOCKING] 5 packages to latest compatible versions
-[UPDATING] just-foo v1.0.0 -> v2.0.0
-[UPDATING] shared v1.0.0 -> v2.0.0
-[UPDATING] transitive-compatible v1.0.0 -> v1.0.1
-[UPDATING] transitive-incompatible v1.0.0 -> v2.0.0
-[UPDATING] ws v1.0.0 -> v2.0.0
+[ERROR] failed to select a version for the requirement `shared = "^0.1"`
+candidate versions found which didn't match: 0.2.0
+location searched: `dummy-registry` index (which is replacing registry `crates-io`)
+required by package `bar v0.0.1 ([ROOT]/foo/bar)`
+    ... which satisfies path dependency `bar` (locked to 0.0.1) of package `foo v0.0.1 ([ROOT]/foo)`
+perhaps a crate was updated and forgotten to be re-vendored?
+
+"#]])
+        .run();
+
+    let foo_manifest_after = p.read_file("Cargo.toml");
+    assert_e2e().eq(&foo_manifest_after, foo_manifest);
+
+    let bar_manifest_after = p.read_file("bar/Cargo.toml");
+    assert_e2e().eq(&bar_manifest_after, bar_manifest);
+}
+
+#[cargo_test]
+fn update_precise_breaking_spec_version() {
+    Package::new("incompatible", "0.1.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                # Check if formatting is preserved
+
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                incompatible  =  "0.1"  # Comment
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("incompatible", "0.3.0").publish();
+    Package::new("incompatible", "0.3.1").publish();
+
+    // No spec
+    p.cargo("update -Zupdate-precise-breaking -p incompatible --precise 0.3.1")
+        .masquerade_as_nightly_cargo(&[])
+        .with_stderr_data(str![[r#"
+[UPGRADING] incompatible ^0.1 -> ^0.3
+[UPDATING] `[..]` index
+[UPDATING] incompatible v0.1.0 -> v0.3.1
+
+"#]])
+        .run();
+
+    // Invalid spec
+    p.cargo("update -Zupdate-precise-breaking -p incompatible@foo --precise 0.3.0")
+        .masquerade_as_nightly_cargo(&[])
+        .with_status(101)
+        .with_stderr_data(str![[r#"
+[ERROR] expected a version like "1.32"
```

Comment:
In case my comment gets lost track of on a merged pr

Hadn't notice this before.  This error message is bad.  We should at least be calling out that we are parsing a package id spec

Existing behavior:
```console
$ cargo update clap@foo
error: invalid package ID specification: `clap@foo`

Caused by:
  expected a version like "1.32"
```
(again, task is list is fine)

Reply:
Added to list.